src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+<?php
+namespace App\Api\Test;
+
+use ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Client as TestClient;
+use App\Api\Test\BaseApiTestCase;
+use App\Entity\Client;
+use App\Entity\Script;
+
+class ClientTest extends BaseApiTestCase
+{
+    public function testCreateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testCreateClientAllParameters(): void 
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $scriptId = $this->getScriptId($httpClient, 'script_all_true');
+        $clientJson = ClientMother::withAllParameters(
+            $clientName, 
+            1, 
+            "some description", 
+            false, 
+            2, 
+            [$scriptId], 
+            [$scriptId],
+            -1, 
+            "rsync long arguments", 
+            "rsync short arguments", 
+            "ssh arguments", 
+            "root@172.0.0.1"
+        );
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+    public function testCreateClientInvalidMaxParallelJobs(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withMaxParallelJobs($clientName, -1);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/ConstraintViolationList',
+            '@type' => 'ConstraintViolationList',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'maxParallelJobs: Max parallel jobs value must be a positive integer',
+        ]);
+    }
+
+    public function testCreateClientInvalidName(): void
```

Comment:
We are not testing for an invalid client name, we are testing for a repeated client name. Please use a more suitable method name.